dask/bytes/local.py

```diff
@@ -82,19 +82,30 @@ def read_block_from_file(path, offset, length, delimiter, compression):
     return result
 
 
-def open_files(path):
+def open_files(path, mode='rb'):
     """ Open many files.  Return delayed objects.
 
     See Also
     --------
     dask.bytes.core.open_files: User function
     """
-    myopen = delayed(open)
-    filepaths = sorted(glob(path))
-    return [myopen(_path, mode='rb',
-                   dask_key_name='open-%s' % tokenize(_path,
-                                                      os.path.getmtime(_path)))
-            for _path in filepaths]
+    if "*" in path:
+        filepaths = sorted(glob(path))
+    elif isinstance(path, str):
+        filepaths = [path]
+    else:
+        filepaths = path
+    if 'r' in mode:
+        myopen = delayed(open)
+        return [myopen(_path, mode=mode,
+                       dask_key_name='open-%s' % tokenize(_path,
+                                                          os.path.getmtime(_path)))
+                for _path in filepaths]
+    else:
+        myopen = delayed(open, pure=False)
+        return [myopen(_path, mode=mode,
+                       dask_key_name='open-%s' % _path)
```

Comment:
There is no reason or the separate `myopen = delayed(open, pure=False)` if you are using keys explicitly.